### PR TITLE
linux build flags

### PR DESCRIPTION
**Pull request by abe-winter**
Add defaults for building on linux. Mac build should :crossed_fingers: be unaffected

### CODE REVIEW

**Review by seanavery — APPROVED** [review]
LGTM! This should work for CI setup as well : )

**Comment by seanavery** [comment]
Looks like I am getting error install go deps on mac. 
```
go.viam.com/rdk/logging
# go.viam.com/rdk/logging
../rdk-droid/logging/net_appender.go:66:24: undefined: apppb.LogEntry
../rdk-droid/logging/net_appender.go:110:16: undefined: apppb.LogEntry
../rdk-droid/logging/net_appender.go:157:51: undefined: apppb.LogEntry
../rdk-droid/logging/net_appender.go:168:55: undefined: apppb.LogEntry
../rdk-droid/logging/net_appender.go:262:51: undefined: apppb.LogEntry
go.viam.com/utils/artifact
make: *** [build-binary] Error 1
```

**Comment by abe-winter** [comment]
guessing I broke the mac build -- thanks for testing

I'm doing APK testing on mac today, will debug this branch

**Comment by abe-winter** [comment]
LogEntry moved recently, it's also possible something in your local dev setup too old or too new

File: go.mod [hunk 1]
```diff
@@ -2,7 +2,7 @@ module github.com/seanavery/viam-droid-cam
 
 go 1.21.5
 
-require go.viam.com/rdk v0.19.1
+require go.viam.com/rdk v0.20.1-0.20240209172210-8dc034cf4d2a
```
Comment (seanavery) [inline go.mod:5]: [note] `v0.20.0` did not include CGO changes. So using more recent commit. Builds working on macos.

File: go.mod [hunk 2]
```diff
@@ -146,5 +146,3 @@ require (
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
 )
-
-replace go.viam.com/rdk => ../rdk-droid
```
Comment (seanavery) [inline go.mod:150]: [note] Removed hardcoded replace.

**Comment by seanavery** [comment]
@abe-winter If this is working on your end w/ linux, we should be good to merge.

File: Makefile [hunk 3]
```diff
@@ -1,15 +1,26 @@
-NDK_ROOT := $(HOME)/Library/Android/sdk/ndk/26.1.10909125
-APP_ROOT := $(HOME)/AndroidStudioProjects/DroidCamera
+UNAME=$(shell uname)
+ifeq ($(UNAME),Linux)
+	NDK_ROOT ?= $(HOME)/Android/Sdk/ndk/26.1.10909125
+	HOST_OS ?= linux
+	GOARCH := amd64
+	CC_ARCH ?= x86_64
```
Comment (seanavery) [inline Makefile:6]: [q] Even if we are on x86 linux machine, we still want to target aarch64 android right?

Comment (abe-winter) [reply Makefile:6]: on intel laptops they give you an intel emulator so that CPU can run it natively

(you may need to override this if you add CI)

**Comment by abe-winter** [comment]
works on linux, merging

**Pull request merged**